icu4c/source/common/BUILD

```diff
@@ -0,0 +1,361 @@
+    srcs = [
+        "utypes.cpp",
+    ],
+    includes = ["."],
+    deps = [
+        ":headers",
+    ],
+    local_defines = [
+        "U_COMMON_IMPLEMENTATION",
+    ],
+)
+
+cc_library(
+    name = "uniset",
+    srcs = [
+        "uniset.cpp",
+        "unifilt.cpp",
+        "unisetspan.cpp",
+        "bmpset.cpp",
+        "util.cpp",
+        "unifunct.cpp",
+        "usetiter.cpp",
+    ],
+    includes = ["."],
+    deps = [
+        ":patternprops",
+        ":uvector",
+        ":headers",
+    ],
+    local_defines = [
+        "U_COMMON_IMPLEMENTATION",
+    ],
+)
+
+cc_library(
+    name = "patternprops",
+    srcs = [
+        "patternprops.cpp",
+    ],
+    includes = ["."],
+    deps = [
+        ":headers",
+    ],
+    local_defines = [
+        "U_COMMON_IMPLEMENTATION",
+    ],
+)
+
+cc_library(
+    name = "propsvec",
+    srcs = [
+        "propsvec.cpp",
+    ],
+    includes = ["."],
+    deps = [
+        ":sort",
+        ":utrie2_builder",
+        ":headers",
+    ],
+    local_defines = [
+        "U_COMMON_IMPLEMENTATION",
+    ],
+)
+
+cc_library(
+    name = "propname",
+    srcs = [
+        "propname.cpp",
+    ],
+    includes = ["."],
+    deps = [
+        ":bytestrie",
+        ":headers",
+    ],
+    local_defines = [
+        "U_COMMON_IMPLEMENTATION",
+    ],
+)
+
+# Note: cc_library target name "uvector32" matches dependencies.txt group name,
+# but filename "uvectr32.*" differs in spelling.
+cc_library(
+    name = "uvector32",
+    srcs = [
+        "uvectr32.cpp",
+    ],
+    includes = ["."],
+    deps = [
+        ":platform",
+        ":headers",
+    ],
+    local_defines = [
+        "U_COMMON_IMPLEMENTATION",
+    ],
+)
+
+cc_library(
+    name = "sort",
+    srcs = [
+        "uarrsort.cpp",
+    ],
+    includes = ["."],
+    deps = [
+        ":headers",
+    ],
+    local_defines = [
+        "U_COMMON_IMPLEMENTATION",
+    ],
+)
+
+cc_library(
+    name = "uvector",
+    srcs = [
+        "uvector.cpp",
+    ],
+    includes = ["."],
+    deps = [
+        ":platform",
+        ":sort",
+    ],
+    local_defines = [
+        "U_COMMON_IMPLEMENTATION",
+    ],
+)
+
+
+# Building this target will indirectly cause the creation of a header file that
+# integrates normalization data. This header file is generated by a genrule
+# target that invokes the gennorm2 binary.
+cc_library(
+    name = "normalizer2",
+    srcs = [
+        "normalizer2.cpp",
+        "normalizer2impl.cpp",
+    ],
+    includes = ["."],
+    hdrs = [
+        "normalizer2impl.h",
+        "norm2_nfc_data.h",
+    ],
+    deps = [
+        ":headers",
+    ],
+    local_defines = [
+        "U_COMMON_IMPLEMENTATION",  
+    ],
+)
+
+
+
+# Note: Bazel has chosen specifically to disallow any official support to break
+# through these constraints that would copying to or generating to source
+# directories: https://stackoverflow.com/questions/51283183/how-to-generate-files-in-source-folder-using-bazel
```

Comment:
```suggestion
# Note: Bazel has chosen specifically to disallow any official support for breaking
# through constraints that would allow copying to or generating to source
# directories: https://stackoverflow.com/questions/51283183/how-to-generate-files-in-source-folder-using-bazel
```